cns/restserver/api.go

```diff
@@ -899,6 +902,94 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
+	logger.Printf("[Azure CNS] getOrRefreshNetworkContainers")
+
+	switch r.Method {
+	case "GET":
+		// return all NCs in CNS
+		networkContainersDetails := make(map[string]cns.ContainerStatus)
+		for ncID := range service.state.ContainerStatus {
+			networkContainersDetails[ncID] = storeNcStatus(service.state.ContainerStatus[ncID])
+		}
+
+		response := cns.GetOrRefreshNetworkContainersResponse{
+			NetworkContainersDetails: networkContainersDetails,
+			Response: cns.Response{
+				ReturnCode: types.Success,
+			},
+		}
+		err := service.Listener.Encode(w, &response)
+		logger.Response(service.Name, response, response.Response.ReturnCode, err)
+
+	case "POST":
+		var req cns.GetOrRefreshNetworkContainersRequest
+		err := service.Listener.Decode(w, r, &req)
+		logger.Request(service.Name, &req, err)
+		if err != nil {
+			logger.Printf("[Azure CNS] getOrRefreshNetworkContainers failed with error: %s", err.Error())
+			return
+		}
+
+		returnCode := types.Success
+		returnMessage := ""
+		for i := 0; i <= len(req.CreateNetworkContainerRequests); i++ {
+			body, encodeErr := json.Marshal(req.CreateNetworkContainerRequests[i])
+			if encodeErr != nil {
+				logger.Printf("[Azure CNS] Create Network Container failed with error: %s", err.Error())
+				return
+			}
+			createNcRequest, requestErr := http.NewRequestWithContext(context.Background(), http.MethodPost, "", bytes.NewBuffer(body))
+			if requestErr != nil {
+				logger.Printf("[Azure CNS] Create Network Container failed with error: %s", err.Error())
+				return
+			}
+			createNcRequest.Header.Set(common.ContentType, common.JsonContent)
+			writer := httptest.NewRecorder()
+			service.createOrUpdateNetworkContainer(writer, createNcRequest)
+
+			var resp cns.CreateNetworkContainerResponse
+			if err = json.Unmarshal(writer.Body.Bytes(), &resp); err != nil || resp.Response.ReturnCode != types.Success {
+				returnCode = types.UnexpectedError
+				returnMessage = fmt.Sprintf("[Azure CNS] Create Network Container failed with error: %s", err.Error())
+				break
+			}
+		}
+
+		response := cns.GetOrRefreshNetworkContainersResponse{
+			Response: cns.Response{
+				ReturnCode: returnCode,
+				Message:    returnMessage,
+			},
+		}
+		err = service.Listener.Encode(w, &response)
+
+		if returnCode == types.Success {
+			service.Lock()
+			saveStateErr := service.saveState()
+			service.Unlock()
+			if saveStateErr != nil {
+				logger.Printf("[Azure CNS] Create Network Container failed with error: %s", err.Error())
+				return
+			}
+		}
+		logger.Response(service.Name, response, response.Response.ReturnCode, err)
+
+	default:
+		response := cns.GetOrRefreshNetworkContainersResponse{
+			Response: cns.Response{
+				ReturnCode: types.InvalidParameter,
+				Message:    "[Azure CNS] Error. getOrRefreshNetworkContainers did not receive a GET or POST.",
+			},
+		}
+		err := service.Listener.Encode(w, &response)
+		logger.Response(service.Name, response, response.Response.ReturnCode, err)
```

Comment:
let's just return an http.MethodNotAllowed instead of this struct

Reply:
Fixed.